experimental/zerotrie/src/varint.rs

```diff
@@ -0,0 +1,405 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Varint spec for ZeroTrie:
+//!
+//! - Lead byte: top 2 bits are trie metadata; third is varint extender; rest is value
+//! - Trail bytes: top bit is varint extender; add rest to current value * 2^7
+//! - Add the "latent value" to the final result: (1<<5) + (1<<7) + (1<<14) + ...
+
+use crate::builder::konst::ConstArrayBuilder;
+
+#[cfg(feature = "alloc")]
+use crate::builder::nonconst::TrieBuilderStore;
+
+/// Reads a varint with 2 bits of metadata in the lead byte.
+pub const fn read_varint(start: u8, remainder: &[u8]) -> Option<(usize, &[u8])> {
+    let mut value = (start & 0b00011111) as usize;
+    let mut remainder = remainder;
+    if (start & 0b00100000) != 0 {
```

Comment:
I prefer the bytes inline rather than trying to come up with good names for the constants and then putting them far away from where they are being used